src/main/scala/com/sksamuel/scapegoat/inspections/collections/CollectionIndexOnNonIndexedSeq.scala

```diff
@@ -0,0 +1,29 @@
+package com.sksamuel.scapegoat.inspections.collections
+
+import com.sksamuel.scapegoat._
+
+/** @author Josh Rosen */
+class CollectionIndexOnNonIndexedSeq extends Inspection {
+
+  def inspector(context: InspectionContext): Inspector = new Inspector(context) {
+    override def postTyperTraverser = Some apply new context.Traverser {
+
+      import context.global._
+
+      private def isSeq(t: Tree) = t.tpe <:< typeOf[Seq[_]]
+      private def isIndexedSeq(t: Tree) = t.tpe <:< typeOf[IndexedSeq[_]]
+
+      override def inspect(tree: Tree): Unit = {
+        tree match {
+          case Apply(Select(lhs, TermName("apply")), List(_)) if isSeq(lhs) && !isIndexedSeq(lhs) =>
```

Comment:
It's probably worth ignoring the `apply(0)` / `apply(constant)` case (like I did in my `linter` implementation.